app/Services/AuthenticationService.php

```diff
@@ -4,44 +4,75 @@
 
 namespace Brewmap\Services;
 
+use Brewmap\Eloquent\SocialProfile;
 use Brewmap\Eloquent\User;
+use Illuminate\Contracts\Hashing\Hasher;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Illuminate\Http\Exceptions\HttpResponseException;
 use Illuminate\Http\Request;
-use Illuminate\Support\Facades\Hash;
-use Laravel\Socialite\Contracts\User as FacebookUser;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Schema;
+use Laravel\Socialite\Contracts\User as SocialUser;
 
 class AuthenticationService
 {
+    protected Hasher $hash;
+    public function __construct(Hasher $hash)
+    {
+        $this->hash = $hash;
+    }
+
     /**
      * @throws HttpResponseException
      */
     public function login(Request $request): string
     {
         $user = User::where("email", $request->email)->first();
-
-        if ($user === null || !Hash::check($request->password, $user->password)) {
-            throw new HttpResponseException(response()->json(["message" => __("auth.credentialsMismatch")], 401));
+        if ($user === null || !$this->hash->check($request->password, $user->password)) {
+            throw new HttpResponseException(response()->json(["message" => __("auth.credentialsMismatch")], Response::HTTP_UNAUTHORIZED));
         }
         return $user->createToken($user->email)->plainTextToken;
     }
 
     public function register(array $input): void
     {
-        $input["password"] = Hash::make($input["password"]);
+        $input["password"] = $this->hash->make($input["password"]);
         $user = User::create($input);
         $user->save();
     }
 
-    public function authorizeByFacebook(FacebookUser $facebookUser): string
+    /**
+     * @throws HttpResponseException
+     */
+    public function getTokenBySocialLogin(SocialUser $socialUser, string $socialProviderName): string
     {
-        $user = User::where("facebook_id", $facebookUser->getId())->first();
-        if (!$user) {
+        if (!$this->isSocialProviderConfigured($socialProviderName)) {
+            throw new HttpResponseException(response()->json(["message" => __("auth.socialServiceError")], Response::HTTP_NOT_IMPLEMENTED));
```

Comment:
I would rather create new exception like `SocialProviderNotConfiguredException` with `__("auth.socialServiceError"` as `$message` property and `501` as `$code`, throw it here and handle in exception handler.